pkg/authorization/registry/localsubjectaccessreview/rest_test.go

```diff
@@ -152,6 +169,92 @@ func TestErrors(t *testing.T) {
 	test.runTest(t)
 }
 
+func TestRegularWithScopes(t *testing.T) {
+	test := &subjectAccessTest{
+		authorizer: &testAuthorizer{
+			allowed: true,
+			reason:  "because good things",
+		},
+		reviewRequest: &authorizationapi.LocalSubjectAccessReview{
+			Action: authorizationapi.AuthorizationAttributes{
+				Namespace: "unittest",
+				Verb:      "delete",
+				Resource:  "deploymentConfigs",
+			},
+			Groups: sets.NewString("not-master"),
```

Comment:
> How is "different" enabled to impersonate "not-master"? Where exactly in the code is it happening?

It's not impersonating.  This is the normal SAR path, so "different" is just asking about whether "not-master" can do something.

For this particular test, its allowed to check that because we've bypassed that level of authorization for the unit test.